File: Problems/No.1557_Minimum_Number_of_Vertices_to_Reach_All_Nodes/minimum_number_of_vertices_to_reach_all_nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        
        // parameters
        std::vector<int> starting_nodes;
        std::vector<bool> node_no_InDegree(n, true);

        // start computing
        for (auto &edge: edges) node_no_InDegree[edge[1]] = false;
        for (int i=0; i<n; i++) 
            if (node_no_InDegree[i] == true) starting_nodes.push_back(i);

        return starting_nodes;

    }
};

File: Problems/No.1557_Minimum_Number_of_Vertices_to_Reach_All_Nodes/minimum_number_of_vertices_to_reach_all_nodes.py
class Solution(object):
    def findSmallestSetOfVertices(self, n, edges):
        """
        :type n: int
        :type edges: List[List[int]]
        :rtype: List[int]
        """

        # parameters
        starting_nodes = []
        node_no_InDegree = [True]*n

        # start computing
        for [outN, inN] in edges:
            node_no_InDegree[inN] = False
        for i in range(n):
            if node_no_InDegree[i]:
                starting_nodes.append(i)

        return starting_nodes
